test/Microsoft.Extensions.Primitives.Tests/StringSegmentTest.cs

```diff
@@ -21,6 +21,18 @@ public void StringSegment_StringCtor_AllowsNullBuffers()
         }
 
         [Fact]
+        public void StringSegmentConstructor_AllowsEmptyBuffers()
+        {
+            // Arrange & Act
+            var segment = new StringSegment("", 0, 0);
```

Comment:
In addition to checking `string.Empty` shouldn't this also test `new StringSegment("abcde", 3, 0)` and similar?